### PR TITLE
Backport frontend-tiles package

### PR DESCRIPTION
**Pull request by pmconne**
The @itwin/frontend-tiles package was introduced on master for inclusion in 4.0. However, the product team most urgently requesting the new features it provides has many dependencies, most of which it does not own, which would all need to be upgraded to 4.0 before the product could integrate+test the new package.

To accelearate things, backport to 3.7 so that (hopefully) only the product itself will need to upgrade.

Backported #5170, #5213, #5226, and #5266.

### CODE REVIEW

File: extensions/frontend-tiles/package.json [hunk 1]
```diff
@@ -0,0 +1,60 @@
+{
+  "name": "@itwin/frontend-tiles",
+  "version": "4.0.0-dev.45",
```
Comment (aruniverse) [inline extensions/frontend-tiles/package.json:3]:
```suggestion
  "version": "3.7.0-dev.5",
```

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -104,6 +106,7 @@ rush.json       @iTwin/itwinjs-core-admins
 /core/frontend                  @kabentley @bbastings @iTwin/itwinjs-core-display
 /core/frontend/src/extension    @calebmshafer @wgoehrig @aruniverse @kckst8 @johnnyd710
 /core/frontend-devtools         @iTwin/itwinjs-core-display @Ellord207
+/core/frontend-tiles            @iTwin/itwinjs-core-display @danieliborra
```
Comment (aruniverse) [inline .github/CODEOWNERS:109]:
```suggestion
/extensions/frontend-tiles            @iTwin/itwinjs-core-display @danieliborra
```

Comment (pmconne) [reply .github/CODEOWNERS:109]: Thanks for spotting that.

**Review by markschlosseratbentley — APPROVED** [review]

File: .github/CODEOWNERS (hunk 2 above)
Comment (aruniverse) [reply .github/CODEOWNERS]: might need to move this to LN 139

**Pull request merged**